frontend/packages/helm-plugin/src/actions/providers.ts

```diff
@@ -65,3 +74,19 @@ export const useTopologyActionProvider = ({
     return [[], true, undefined];
   }, [connectorSource, element, namespace]);
 };
+
+export const useHelmChartRepositoryActions = (resource: K8sResourceKind) => {
+  const [kindObj, inFlight] = useK8sModel(referenceFor(resource));
+  const { t } = useTranslation();
+  const actions = React.useMemo(() => {
+    let commonActions = getCommonResourceActions(kindObj, resource);
+    const index = commonActions.findIndex((action: Action) => action.id === 'edit-resource');
+    if (index) {
+      commonActions = commonActions.filter((action: Action) => action.id !== 'edit-resource');
+      commonActions.splice(index, 0, editChartRepository(kindObj, resource, t));
+    }
+    return commonActions;
```

Comment:
Do you want check here if the index exist?

Array.findIndex returns -1 if it doesn't exist and would run your if statement.

And it returns 0 if the first action is edit-resource and this would jump over your if statement.

You should also be very careful with Array.splice as it doesn't return a new array. That's only okay here because of the filter one line above. I always recommend to use Array.slice instead also it's not required here.

```suggestion
    let commonActions = getCommonResourceActions(kindObj, resource);
    const index = commonActions.findIndex((action: Action) => action.id === 'edit-resource');
    if (index => 0) {
      commonActions = commonActions.filter((action: Action) => action.id !== 'edit-resource');
      commonActions = commonActions.slice(index, 0, editChartRepository(kindObj, resource, t));
    }
    return commonActions;
```